compiler/src/iree/compiler/Codegen/Common/GPU/GPUDistributionPatterns.cpp

```diff
@@ -727,8 +836,8 @@ void populateGPUDistributionLayoutAttrPatterns(Value laneId,
   patterns
       .add<DistributeTransferReadLayoutAttr, DistributeTransferWriteLayoutAttr>(
           patterns.getContext(), laneId);
-  patterns.add<DistributeBroadcastLayoutAttr, DistributeTranspose>(
-      patterns.getContext());
+  patterns.add<DistributeBroadcastLayoutAttr, DistributeTranspose,
+               DistributeLayoutConflictResolutions>(patterns.getContext());
```

Comment:
This should be in its own populate pattern because it's not a generally applicable pattern. It only works when the user knows its only used for elementwise or specific way of doing matrix multiplication.

Reply:
The newly added populate function should also have a TODO that we may need a new op / analysis to determine when this is safe

Reply:
sounds good, I have add a new PR to address this